### PR TITLE
#165020158 Configure CircleCI to run tests 

### PR DESCRIPTION
**Pull request by niyobobo**
### What does this PR do?
Configuration of the automated test server 
### Description of Task to be completed?
- Run automated tests using CircleCI
- Send test coverage to CodeClimate
### How should this be manually tested?
- Checking CircleCi badge on the README
### Any background context you want to provide?
- Test status shows that it fails because we have no tests yet.
### What are the relevant pivotal tracker stories?
[#165020158](https://www.pivotaltracker.com/n/projects/2321840/stories/165020158)

### CODE REVIEW

**Review by avpaul — APPROVED** [review]
Implemented well! 👍 for adding packages caching!

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            chmod +x ./cc-test-reporter
+    
+      # Save the dependency cache
+      - save_cache:
+          key: dependency-cache-{{ checksum "package.json" }}-{{ checksum "package-lock.json" }}
+          paths:
+            - ./node_modules
+      
+      # Run tests and coverage
+      - run:
```
Comment (ezkemboi) [inline .circleci/config.yml:28]: This is good work, let's work on PT story [165020157](https://www.pivotaltracker.com/story/show/165020157). 
And let's work on making it build successfully and we will be good to merge it.

**Review by ezkemboi — COMMENTED** [review]
Good work but left a comment.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+version: 2
```
Comment (aaronsekisambu) [inline .circleci/config.yml:1]: @niyobobo Thank you this is a great approach.

However, I have a few concerns.  You have limited the circle Ci version to 2. I have seen that there is a new version which is stable [2.17.1](https://circleci.com/server/changelog/)

**Review by aaronsekisambu — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (niyobobo) [reply .circleci/config.yml:28]: @me-x-mi Thank you. Now the feature is done.

File: .circleci/config.yml (hunk 2 above)
Comment (niyobobo) [reply .circleci/config.yml:1]: Thank you @aaronsekisambu for your feedback. 

In semantic versioning style `[Major.minor.patch]` when you specify the major index it checks the latest stable version, not a fixed version. This means CircleCI will use the latest version started by major two.

**Comment by ezkemboi** [comment]
This PR is good to be merged, but it needs one more job, 
Get the latest changes from the develop branch. 
And get the tests passing in circle CI.

**Comment by niyobobo** [comment]
> This PR is good to be merged, but it needs one more job,
> Get the latest changes from the develop branch.
> And get the tests passing in circle CI.

Let me work on it.

**Pull request merged**